apps/api/src/rest/routes/catalogue.ts

```diff
@@ -0,0 +1,67 @@
+import { defaultHook } from "$hooks";
+import { productionCache } from "$middleware";
+import {
+  errorSchema,
+  responseSchema,
+  sampleProgramsQuerySchema,
+  sampleProgramsResponseSchema,
+} from "$schema";
+import { ProgramsService } from "$services";
+import { OpenAPIHono, createRoute } from "@hono/zod-openapi";
+import { database } from "@packages/db";
+
+const catalogueRouter = new OpenAPIHono<{ Bindings: Env }>({ defaultHook });
+
+const sampleProgramsRoute = createRoute({
+  summary: "Sample Programs",
+  operationId: "getSamplePrograms",
+  tags: ["Catalogue"],
+  method: "get",
+  path: "/sample-programs",
+  description:
+    "List all available sample programs in UCI's current catalogue. Course IDs are matched on a best-effort basis, so some rows may not be a courseID.",
+  request: { query: sampleProgramsQuerySchema },
+  responses: {
+    200: {
+      content: {
+        "application/json": {
+          schema: responseSchema(sampleProgramsResponseSchema),
+        },
+      },
+      description: "Successful operation",
+    },
+    404: {
+      content: { "application/json": { schema: errorSchema } },
+      description: "Sample program data not found",
+    },
+    500: {
+      content: { "application/json": { schema: errorSchema } },
+      description: "Server error occurred",
+    },
+  },
+});
+
+catalogueRouter.get(
+  "*",
+  productionCache({ cacheName: "anteater-api", cacheControl: "max-age=86400" }),
+);
+
+catalogueRouter.openapi(sampleProgramsRoute, async (c) => {
+  const query = c.req.valid("query");
+  const service = new ProgramsService(database(c.env.DB.connectionString));
+  const res = await service.getSamplePrograms(query);
+
+  if (query?.id && !res.length) {
```

Comment:
The fact that you check for an ID here and not in GraphQL suggests that you intended them to have parity. You should decide on the behavior.